_posts/2021-05-04-how-to-update-your-ref-arch.adoc

```diff
@@ -0,0 +1,490 @@
+* Rely on Gruntwork to provide feature updates and bug fixes to the services.
+
+Though the Service Catalog fully replaces the services defined in your `infrastructure-modules` repository, there are
+some differences that require you to make modifications to Terragrunt configurations in your `infrastructure-live`
+repository before you can start using the Service Catalog.
+
+
+[[deployment_walkthrough]]
+== Deployment Walkthrough
+
+Your goal in this walkthrough is to replace references to `infrastructure-modules` with `terraform-aws-service-catalog`,
+within your `infrastructure-live` repo. You will follow a two-step process:
+
+* Step 1: Prepare your Terragrunt `infrastructure-live` configuration to support the Service Catalog
+* Step 2: Update each Terragrunt live configuration to use the Service Catalog
+
+IMPORTANT: This guide assumes a minimum terragrunt version of
+https://github.com/gruntwork-io/terragrunt/releases/tag/v0.28.18[0.28.18]. If you are using an older terragrunt version,
+be sure to update to at least this version first.
+
+=== Step 1: Prepare your Terragrunt configuration to support the Service Catalog
+
+The Terraform modules in the Service Catalog are missing some required blocks for Terraform to operate (e.g., the
+`provider` and `terraform` state backend blocks). This is by design, to allow the modules to be flexibly used in
+different contexts (unlike the `infrastructure-modules` modules which are only intended to be used in the Terragrunt
+context). As such, you need to inject these required blocks to use advanced features of Terragrunt. Here we define a new
+root `terragrunt.hcl` configuration that includes these directives so that we can continue to use the original,
+non-Service Catalog based live configuration.
+
+For reference, you can download the _<insert link to the
+https://github.com/gruntwork-io/prototypes/tree/master/ra-folder-structure[prototypes/ra-folder-structure] release>_
+folder which contains all the following files in their respective locations in the folder structure. You'll have to copy
+these files into your own `infrastructure-live` repository.
+
+[arabic]
+. Create a new file `terragrunt_service_catalog.hcl` at the root of the `infrastructure-live` repository. We name this
+  differently from the existing `terragrunt.hcl` so that the modules that haven't migrated yet won't be affected.
+. Insert the following contents:
++
+....
+# ---------------------------------------------------------------------------------------------------------------------
+# TERRAGRUNT CONFIGURATION
+# ---------------------------------------------------------------------------------------------------------------------
+
+locals {
+  common_vars  = read_terragrunt_config("${get_terragrunt_dir()}/common.hcl")
+  account_vars = read_terragrunt_config(find_in_parent_folders("account.hcl"))
+
+  # Optional: Update to use HCL instead of YAML
+  region_vars  = yamldecode(file(find_in_parent_folders("region.yaml")))
```

Comment:
this is a pretty easy step to follow, so just make it non optional?

Reply:
Same feelings as above. Even if it's easy, it's not required, so I'm leaning toward keeping it optional.